src/TableView.jl

```diff
@@ -1,119 +1,152 @@
 module TableView
+using Tables, TableTraits, IteratorInterfaceExtensions
+using WebIO, JSExpr, JSON, Dates
+using Observables: @map
 
-using WebIO
-using JSExpr
-using JuliaDB
-using DataValues
+const ag_grid_imports = []
 
-import JuliaDB: DNDSparse, DNextTable, NextTable
-
-function JuliaDB.subtable(t::DNextTable, r)
-    table(collect(rows(t)[r]), pkey=t.pkey)
+function __init__()
+    empty!(ag_grid_imports)
+    for f in ["ag-grid.js", "ag-grid.css", "ag-grid-light.css", "ag-grid-dark.css"]
+        push!(ag_grid_imports, normpath(joinpath(@__DIR__, "..", "deps", "ag-grid", f)))
+    end
 end
 
-showna(xs) = xs
-function showna(xs::AbstractArray{T}) where {T<:DataValue}
-    map(xs) do x
-        isnull(x) ? "NA" : get(x)
+function showtable(x; dark = false)
+    iit = TableTraits.isiterabletable(x)
+    if Tables.istable(typeof(x))
+        return _showtable(x, dark)
+    elseif ismissing(iit) || iit
+        it = IteratorInterfaceExtensions.getiterator(x)
+        return _showtable(Tables.DataValueUnwrapper(it), dark)
     end
+    throw(ArgumentError("Argument is not a table."))
 end
 
-function showna(xs::Columns)
-    rows(map(showna, columns(xs)))
+function _showtable(table, dark)
+    tablelength = Base.IteratorSize(table) == Base.HasLength() ? length(Tables.rows(table)) : nothing
+
+    rows = Tables.rows(table)
+    schema = Tables.schema(table)
+    if schema === nothing
+        types = []
+        for (i, c) in enumerate(Tables.eachcolumn(first(rows)))
+            push!(types, typeof(c))
+        end
+        names = collect(propertynames(first(rows)))
+    else
+        names = schema.names
+        types = schema.types
+    end
+    w = Scope(imports = ag_grid_imports)
```

Comment:
@shashi Is there a way to directly style the `div` introduced by the `Scope`? This results in something like
```
<div class="webio-mountpoint interactbulma" data-webio-mountpoint="13462147745702835718">
  <div class="webio-scope" data-webio-scope-id="scope-4acc84b9-6adc-4a65-8534-e785737a7c2d">
    <div id="grid" class="ag-theme-balham" style="height: 100%; min-height: 200px; width: 100%;">
```
right now, but I'd also like to give the `Scope` (and the mountpoint, I guess) a style to allow piping through the parent's width and height.

Reply:
We don't have that right now, but the way to do it would be allow `s(style=Dict(:foo => bar))` to work where `s` is a Scope (just like it works for Node)...